pkg/controller/registry/resolver/resolver.go

```diff
@@ -167,42 +167,47 @@ func (resolver *SingleSourceResolver) resolveCRDDescription(sources map[registry
 type MultiSourceResolver struct{}
 
 // ResolveInstallPlan resolves the given InstallPlan with all available sources
-func (resolver *MultiSourceResolver) ResolveInstallPlan(sources map[registry.SourceKey]registry.Source, firstSrcKey registry.SourceKey, catalogLabelKey string, plan *v1alpha1.InstallPlan) ([]v1alpha1.Step, error) {
+func (resolver *MultiSourceResolver) ResolveInstallPlan(sources map[registry.SourceKey]registry.Source, preferredSourceKey registry.SourceKey, catalogLabelKey string, plan *v1alpha1.InstallPlan) ([]v1alpha1.Step, []registry.SourceKey, error) {
 	srm := make(stepResourceMap)
+	var usedSourceKeys []registry.SourceKey
+
 	for _, csvName := range plan.Spec.ClusterServiceVersionNames {
 
 		// Attempt to resolve from the first CatalogSource
-		csvSRM, err := resolver.resolveCSV(sources, firstSrcKey, catalogLabelKey, csvName)
+		csvSRM, used, err := resolver.resolveCSV(sources, preferredSourceKey, catalogLabelKey, plan.Namespace, csvName)
 
 		if err == nil {
 			srm.Combine(csvSRM)
+			usedSourceKeys = append(used, usedSourceKeys...)
 			continue
 		}
 
 		// Attempt to resolve from any other CatalogSource
 		for srcKey := range sources {
```

Comment:
Only resolve from `CatalogSources` in the same namespace of the `InstallPlan` *or* the "global" namespace (`o.namespace`).

Reply:
I reduced the catalog sources snapshot that gets passed in to only contain sources from `plan.Namespace` and `o.namespace`.  That should address this issue.